### DISCUSSION TITLE
reload file from disk automatically

### DISCUSSION THREAD

**Discussion by goyalyashpal** [jupyterlab/jupyterlab#14516] [Q&A] [answered]
_edit: go to https://github.com/jupyterlab/jupyterlab/issues/14619_

reload file from disk automatically

any way to do that? 

this question seems popular on SE as well.

https://stackoverflow.com/questions/56112948/is-there-a-way-to-reload-notebook-from-disk-automatically-and-periodically-wit

https://stackoverflow.com/questions/63364985/auto-reloading-jupyter-notebook-when-file-changed-on-disk

**Comment by davidbrochart** [comment]
It already works in collaborative mode, just `pip install jupyter_collaboration`.

**Reply by goyalyashpal** [reply]
i installed that, and have restarted the jupyter lab server too, have looked under file option and under settings

but couldnt find collaborative mode anywhere so far.

i forgot to check `jupyter-lab --help` though

**Reply by davidbrochart** [reply, marked as answer]
Sorry, I forgot to mention that you need to `pip install --pre jupyterlab`, until JupyterLab 4.0 is released. And it should work out of the box.

**Comment by goyalyashpal** [comment]
how to launch the "pre" versions of the pip apps? `jupyter lab` still 
launches the v3.6

David Brochart wrote:
>
> Sorry, I forgot to mention that you need to |pip install --pre 
> jupyterlab|, until JupyterLab 4.0 is released. And it should work out 
> of the box.Message ID: 
> ***@***.***>
>

**Reply by davidbrochart** [reply]
Maybe start over from a fresh environment?

**Reply by goyalyashpal** [reply]
😢 aah, i am getting a live taste of deps hell

i don't know about environments, i am getting stuck with micromamba installation...
even if i do install mamba, then too, i don't know how to use those..

before this i tried doing just `pip uninstall jupyterlab` then `pip install --pre jupyterlab` and then on launching, it was not fetching the files...

so, dabbled into touching other components - i think it eventually worked out in the end

<details>

```console
$ pip uninstall jupyterlab
  Successfully uninstalled jupyterlab-3.6.3

$ pip install --pre jupyterlab
Successfully installed jupyterlab-4.0.0rc1

$ pip check
cs50 9.2.5 has requirement SQLAlchemy==1.4.46, but you have sqlalchemy 2.0.12.
jupyter-collaboration 1.0.0a8 has requirement jupyter-ydoc<2.0.0,>=1.0.1, but you have jupyter-ydoc 0.2.4.
jupyter-collaboration 1.0.0a8 has requirement ypy-websocket<0.9.0,>=0.8.3, but you have ypy-websocket 0.8.2.

$ pip install -U jupyter-ydoc
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jupyter-collaboration 1.0.0a8 requires ypy-websocket<0.9.0,>=0.8.3, but you have ypy-websocket 0.8.2 which is incompatible.
jupyter-server-ydoc 0.8.0 requires jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2 which is incompatible.
ypy-websocket 0.8.2 requires y-py<0.6.0,>=0.5.3, but you have y-py 0.6.0 which is incompatible.
Successfully installed jupyter-ydoc-1.0.2 y-py-0.6.0

$ pip check
cs50 9.2.5 has requirement SQLAlchemy==1.4.46, but you have sqlalchemy 2.0.12.
jupyter-collaboration 1.0.0a8 has requirement ypy-websocket<0.9.0,>=0.8.3, but you have ypy-websocket 0.8.2.
jupyter-server-ydoc 0.8.0 has requirement jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2.
ypy-websocket 0.8.2 has requirement y-py<0.6.0,>=0.5.3, but you have y-py 0.6.0.

$ pip install -U ypy-websocket
jupyter-collaboration 1.0.0a8 requires ypy-websocket<0.9.0,>=0.8.3, but you have ypy-websocket 0.9.0 which is incompatible.
jupyter-server-ydoc 0.8.0 requires jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2 which is incompatible.
jupyter-server-ydoc 0.8.0 requires ypy-websocket<0.9.0,>=0.8.2, but you have ypy-websocket 0.9.0 which is incompatible.
Successfully installed ypy-websocket-0.9.0

$ pip check
cs50 9.2.5 has requirement SQLAlchemy==1.4.46, but you have sqlalchemy 2.0.12.
jupyter-collaboration 1.0.0a8 has requirement ypy-websocket<0.9.0,>=0.8.3, but you have ypy-websocket 0.9.0.
jupyter-server-ydoc 0.8.0 has requirement jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2.
jupyter-server-ydoc 0.8.0 has requirement ypy-websocket<0.9.0,>=0.8.2, but you have ypy-websocket 0.9.0.

$ pip install -U jupyter-collaboration
Installing collected packages: ypy-websocket
  Attempting uninstall: ypy-websocket
    Found existing installation: ypy-websocket 0.9.0
    Uninstalling ypy-websocket-0.9.0:
      Successfully uninstalled ypy-websocket-0.9.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
jupyter-server-ydoc 0.8.0 requires jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2 which is incompatible.
Successfully installed ypy-websocket-0.8.4

$ pip check
cs50 9.2.5 has requirement SQLAlchemy==1.4.46, but you have sqlalchemy 2.0.12.
jupyter-server-ydoc 0.8.0 has requirement jupyter-ydoc<0.4.0,>=0.2.0, but you have jupyter-ydoc 1.0.2.
```

</details>

**Reply by goyalyashpal** [reply]
~but the thing seems to be - that i still can't find collaborative mode in jupyterlab Version 4.0.0rc1~

ohkay, the (AL) mark in the top right corner was the session name, and the collab icon appeared in left activity bar

**Comment by goyalyashpal** [comment]
so, umh, the .py jupytext notebook (https://github.com/jupyterlab/jupyterlab/issues/14386#issuecomment-1510297203) is not supported this way? 😢 

the normal .ipynb notebooks do auto reload
